### PR TITLE
libaccessom2: allow variants to directly modify CMakeLists.txt

### PR DESCRIPTION
**Pull request by harshula**
Modify CMakeLists.txt directly from the SPD instead of having a two step process, where we have to add the variant to the SPD and then go modify the package's source code repository.

### CODE REVIEW

File: packages/libaccessom2/package.py [hunk 1]
```diff
@@ -38,6 +38,14 @@ class Libaccessom2(CMakePackage):
     def url_for_version(self, version):
         return "https://github.com/ACCESS-NRI/libaccessom2/tarball/{0}".format(version)
 
-    def cmake_args(self):
```
Comment (CodeGat) [inline packages/libaccessom2/package.py:41]: Just a quick check: did anything call this function that would miss it being deleted?

Comment (harshula) [reply packages/libaccessom2/package.py:41]: My understanding is that is a function over-ride mechanism as described in https://spack.readthedocs.io/en/latest/build_systems/cmakepackage.html#adding-flags-to-cmake .

Comment (CodeGat) [reply packages/libaccessom2/package.py:41]: Noted.

**Review by CodeGat — APPROVED** [review]
Talked about this over a call. Approved.

**Pull request merged**